gtd_generative_proof_machine/inc/generative_restriction.h

```diff
@@ -0,0 +1,60 @@
+#ifndef GENERATIVE_RESTRICTION_H
+#define GENERATIVE_RESTRICTION_H
+
+#include "common.h"
+#include "physical_graph.h"
+#include "generative_proof_machine.h"
+
+typedef struct GenerativeRestriction GenerativeRestriction;
+typedef struct intNode intNode;
+typedef struct PathNode PathNode;
+
+typedef struct PathNode
+{
+    int *path;
```

Comment:
good practice is to always specify int size and sign f.e. int32_t, uint8_t from stdint.h (does not have to be changed everywhere now, do not waste time)